python/catkin/find_in_workspaces.py

```diff
@@ -75,6 +75,27 @@ def _get_valid_search_dirs(search_dirs, project):
     return search_dirs
 
 
+def collect_path_data_in_workspaces(workspaces=None,
```

Comment:
Actually this function cannot be resused in `find_in_workspaces()` because `find_in_workspaces()` has `first_matching_workspace_only` arg.
And reusing can cause low performance for it, because this function see all workspaces and cannot stop after first match.